File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/search/repository/SearchBookRepository.java
package com.hgianastasio.biblioulbrav2.core.search.repository;

import com.hgianastasio.biblioulbrav2.core.base.boundaries.OnResultListener;
import com.hgianastasio.biblioulbrav2.core.search.models.SearchModel;
import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult;

import java.io.IOException;

/**
 * Created by heitorgianastasio on 4/24/17.
 */

public interface SearchBookRepository {
    void search(SearchModel model, OnResultListener<SearchResult> onResultListener) throws IOException;
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/search/repository/SearchBookRepository.kt
package com.hgianastasio.biblioulbrav2.core.search.repository

import com.hgianastasio.biblioulbrav2.core.base.boundaries.OnResultListener
import com.hgianastasio.biblioulbrav2.core.search.models.SearchModel
import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult
import java.io.IOException

/**
 * Created by heitorgianastasio on 4/24/17.
 */
interface SearchBookRepository {
    @Throws(IOException::class)
    fun search(model: SearchModel?, onResultListener: OnResultListener<SearchResult>)
}